README.md

```diff
@@ -40,189 +40,113 @@ When using GitLink, the user no longer has to specify symbol servers. The only r
 
-## Linking a context
+When you share your PDB alongside your DLL, your users who debug with Source Server support enabled will automatically be able to step into your source code. 
 
-Once a context is created, the *Linker* class can be used to actually link the files:
+# Troubleshooting
 
-    Linker.Link(context);
+## Source Stepping isn't working
 
-# How to get
+* Visual Studio 2012 needs to run elevated in order to download the source server files
 
-There are three general ways to get GitLink:
+* Specify a value for Visual Studio -> Options -> Debugging -> Symbols -> `Cache Symbols in this directory`
 
-## Get it from GitHub
+![Enabling source server support](doc/images/visualstudio_symbolslocation.png)
 
-The releases will be available as separate executable download on the [releases tab](https://github.com/GitTools/GitLink/releases) of the project.
+## Source Stepping returns HTML
+If your repository is private, you are likely seeing the logon HTML from your git host.
 
-## Get it via Chocolatey
+* Log onto your git host in Internet Explorer
+* Purge your local symbol cache
 
-If you want to install the tool on your (build) computer, the package is available via <a href="https://chocolatey.org/" target="_blank">Chocolatey</a>. To install, use the following command:
+Note that this approach is not guaranteed to work.  Visual Studio needs to authenticate to retrieve the source files
+but does not ask the user for credentials to do so.  There are ways to work around this, but no mechanism is currently
+provided out-of-the-box in *GitLink*.
 
-    choco install gitlink
+Possible workarounds
+* Include a mechanism in the pdb to retrieve credentials (using PowerShell and Windows credentials store) (see [#37](https://github.com/GitTools/GitLink/issues/37))
+* Use a proxy service that does not require authentication (see [#66](https://github.com/GitTools/GitLink/issues/66) and [Source server with Git repository](https://shonnlyga.wordpress.com/2016/05/28/source-server-with-git-repository))
 
-## Get it via NuGet
 
-If you want to reference the assembly to use it in code, the recommended way to get it is via <a href="http://www.nuget.org/" target="_blank">NuGet</a>. 
+# Supported git providers
 
-**Note that getting GitLink via NuGet will add it as a reference to the project**
+GitLink supports the following providers out of the box (will auto-detect based on the url):
 
-# How does it work
+* <a href="https://bitbucket.org/" target="_blank">BitBucket</a>
+* <a href="https://github.com/" target="_blank">GitHub</a>
+* Custom Provider (custom urls)
 
-The SrcSrv tool (Srcsrv.dll) enables a client to retrieve the exact version of the source files that were used to build an application. Because the source code for a module can change between versions and over the course of years, it is important to look at the source code as it existed when the version of the module in question was built.
+Providers currently being worked on:
```

Comment:
Let's rename this to:

"Providers that could be supported with the help of the community"